agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4440,36 +4486,15 @@ beerocks::message::sWifiChannel slave_thread::channel_selection_select_channel()
             continue;
         }
         for (uint8_t i = 0; i < beerocks::message::SUPPORTED_CHANNELS_LENGTH; i++) {
-            auto channel        = hostap_params.supported_channels[i];
-            auto bw             = static_cast<beerocks::eWiFiBandwidth>(channel.channel_bandwidth);
-            auto center_channel = wireless_utils::get_5g_center_channel(channel.channel, bw, true);
-            auto operating_class =
-                wireless_utils::get_operating_class_by_channel(channel.channel, center_channel, bw);
-
-            // Skip channels from other operating classes
-            if (operating_class != preference.oper_class) {
-                continue;
-            }
+            const auto &channel = hostap_params.supported_channels[i];
             // Skip DFS channels
             if (hostap_params.supported_channels[i].is_dfs_channel) {
                 LOG(DEBUG) << "Skip DFS channel " << int(channel.channel);
                 continue;
             }
-            auto is_restricted =
-                std::find_if(preference.channels.begin(), preference.channels.end(),
-                             [&](const beerocks::message::sWifiChannel &wifi_channel) {
-                                 // operating classes 128,129,130 use center channel **unlike the other classes** (See Table E-4 in 802.11 spec)
-                                 if (operating_class == 128 || operating_class == 129 ||
-                                     operating_class == 130) {
-                                     return center_channel == wifi_channel.channel;
-                                 } else {
-                                     return (channel.channel == wifi_channel.channel);
-                                 }
-                             });
             // Skip restricted channels
-            if (is_restricted != preference.channels.end()) {
-                LOG(DEBUG) << "Skip restricted channel " << int(channel.channel)
-                           << " (center channel " << int(center_channel) << ")";
+            if (is_channel_restricted(channel, preference)) {
```

Comment:
So when returning the preference value, you'd check here if it was equal to NON_OPERABLE. And it's trivial to extend it later to multiple preference values.